src/fusor/extract.py

```diff
@@ -3,55 +3,64 @@
 """
 
 import csv
-import logging
 from pathlib import Path
 
 from fusor.fusion_caller_models import (
     JAFFA,
     Arriba,
     Cicero,
     EnFusion,
+    FusionCaller,
     FusionCatcher,
     Genie,
     STARFusion,
 )
 
-_logger = logging.getLogger(__name__)
 
-
-def _check_if_file_exists(path: Path) -> bool:
+def _does_file_exist(path: Path) -> None:
     """Check if fusions file exists
 
     :param path: The path to the file
     :return ``True`` if the file exists, ``False`` if not
     """
     if not path.exists():
         statement = f"{path!s} does not exist"
-        _logger.error(statement)
-        return False
-    return True
+        raise ValueError(statement)
+    return
+
+
+def _process_fusion_caller_rows(
+    path: Path, caller: FusionCaller, column_rename: dict
+) -> list[FusionCaller]:
+    """Convert rows of fusion caller output to Pydantic classes
+
+    :param path: The path to the fusions file
+    :param caller: The name of the fusion caller
+    :param column_rename: A dictionary of column mappings
+    :return: A list of fusions, represented as Pydantic objects
+    """
+    fusions_list = []
+    with path.open() as csvfile:
+        reader = csv.DictReader(csvfile, delimiter="," if caller == JAFFA else "\t")
+        for row in reader:
+            row = {column_rename.get(key, key): value for key, value in row.items()}
+            fusions_list.append(caller(**row))
+    return fusions_list
 
 
-def get_jaffa_records(path: Path) -> list[JAFFA] | None:
+def get_jaffa_records(path: Path) -> list[JAFFA]:
     """Load fusions from JAFFA csv file
 
     :param path: The path to the file of JAFFA fusions
     :return A list of JAFFA objects, or None if the specified file does not exist
     """
-    if not _check_if_file_exists(path):
-        return None
-    fusions_list: list[JAFFA] = []
+    _does_file_exist(path)
```

Comment:
Since we're checking in each, we could move this inside `_process_fusion_caller_rows`